pinot-controller/src/main/java/org/apache/pinot/controller/util/ServerSegmentMetadataReader.java

```diff
@@ -47,6 +61,11 @@ public class ServerSegmentMetadataReader {
   private final Executor _executor;
   private final HttpClientConnectionManager _connectionManager;
 
+  public ServerSegmentMetadataReader() {
+    _executor = Executors.newFixedThreadPool(1);
```

Comment:
Is this enough? Assuming these are quick calls to the server

Reply:
oh good catch. I put 1 while testing it out and left it over. For now, I changed it to `numAvailableProcessor`